src/geometry/types.rs

```diff
@@ -119,8 +119,60 @@ impl Vector {
             dy: self.dy / magnitude,
         })
     }
+
+// Rectangle /////////////////////////////////////////////////////////////////
+pub struct Size{
+    pub width: f32,
+    pub heigh: f32,
+}
+
+pub struct Rectangle {
```

Comment:
I would suggest to rename `Rectangle` to `Rect`. It's always better to use shorter names when possible so that the code remains clean.